### ISSUE TITLE
Declaring tests in a chained fashion has different behaviour when using assertion in @after hook

### ISSUE THREAD

**Issue by frankdejonge** [pestphp/pest#816]
Given this abstract test case

```php
<?php

use PHPUnit\Framework\TestCase;

abstract class AbstractTestCase extends TestCase
{
    private bool $shouldFail = false;

    /**
     * @after
     */
    public function failIfNeeded(): void
    {
        if ($this->shouldFail) {
            self::fail('we should fail');
        } else {
            self::assertTrue(true);;
        }
    }

    /**
     * @before
     */
    public function resetFailureSetup(): void
    {
        $this->shouldFail = false;
    }

    public function setupToFail(): void
    {
        $this->shouldFail = true;
    }
}
```

and this Pest file:

```php
<?php

uses(AbstractTestCase::class);

it('this test will fail', fn () => $this->setupToFail());

it('this test will be marked as incomplete')->setupToFail();

it('this test will not fail', fn () => expect(true)->toBe(true));
```

The first test will fail, the second will be masked as incomplete, and the last one will fail. When doing these kinds of assertions in regular PHPUnit, the tests behave like the first case. It seems like there is a difference in how lifecycle hooks are executed is different depending on the style of writing tests.

I'm using `pestphp/pest:v2.5.3` with `phpunit/phpunit:10.1.2`.

**Comment by frankdejonge** [comment]
Looks like the location of this decision is the culprit: https://github.com/pestphp/pest/blob/2.x/src/Factories/TestCaseMethodFactory.php#L66

**Comment by frankdejonge** [comment]
As a resolution path, I tried adding an @after hook in the generated test class created by the `TestCaseFactory`, however, it seems that `PHPUnit\Framework\Assert:resetCount()` is called before this hook is run. Replacing the incomplete check in the closure from my previous comment results in the test being marked as risky, which doesn't seem like bad behaviour, but there might be context as to why the Pests result is different than that of PHPUnit when it comes to tests without assertions.

**Comment by devajmeireles** [comment]
Hey, @frankdejonge ! Thanks for your report.

I'm checking with the Pest team if we have plans to that.

**Comment by nunomaduro** [comment]
This is fixed on `v2.33.0`
